pkg/live/inventoryrg.go

```diff
@@ -228,20 +230,71 @@ var crdGroupKind = schema.GroupKind{
 	Kind:  "CustomResourceDefinition",
 }
 
-// ResourceGroupCRDApplied returns true if the inventory ResourceGroup
-// CRD is available from the current RESTMapper, or false otherwise.
-func ResourceGroupCRDApplied(factory cmdutil.Factory) bool {
+// ResourceGroupCRDApplied returns (true, true) if the inventory ResourceGroup
+// CRD is available from the current RESTMapper and matches the latest declaration.
+// It returns (false, false) if the inventory ResourceGroup CRD is not installed.
+// It returns (true, false) if the ResourceGroup CRD is installed, but
+// doesn't match the latest declaration.
+func ResourceGroupCRDApplied(factory cmdutil.Factory) (bool, bool) {
 	mapper, err := factory.ToRESTMapper()
 	if err != nil {
 		klog.V(4).Infof("error retrieving RESTMapper when checking ResourceGroup CRD: %s\n", err)
-		return false
+		return false, false
 	}
 	_, err = mapper.RESTMapping(ResourceGroupGVK.GroupKind())
 	if err != nil {
 		klog.V(7).Infof("error retrieving ResourceGroup RESTMapping: %s\n", err)
+		return false, false
+	}
+	return true, resourceGroupCRDMatchesLatest(mapper, factory)
+}
+
+func resourceGroupCRDMatchesLatest(mapper meta.RESTMapper, factory cmdutil.Factory) bool {
+	rgCRDStr, ok := resourceGroupCRDs["v1"]
+	if !ok {
+		klog.V(7).Infof("ResourceGroup CRD version v1 not found")
+		return false
+	}
+	crd, err := stringToUnstructured(rgCRDStr)
+	if err != nil {
+		klog.V(7).Infof("failed to get ResourceGroup CRD from string: %s", err)
+		return false
+	}
+
+	dc, err := factory.DynamicClient()
+	if err != nil {
+		klog.V(7).Infof("error getting the dynamic client: %s\n", err)
+		return false
+	}
+
+	mapping, err := mapper.RESTMapping(crdGroupKind)
+	if err != nil {
+		klog.V(7).Infof("Failed to get mapping of CRD type: %s", err)
+		return false
+	}
+
+	liveCRD, err := dc.Resource(mapping.Resource).Get(context.TODO(), "resourcegroups.kpt.dev", metav1.GetOptions{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "apiextensions.k8s.io/v1",
```

Comment:
Do we need to handle clusters that doesn't have the v1 version of the CRD resource type?

Reply:
I don't think we need to continue providing support for v1beta1 CRD. The v1 version of the CRD is available since 1.16, which is quite a few versions before the current one 1.23.

Reply:
I'm not sure we are ready to assume all cluster have the v1 version yet. In any case, we should do that in a separate PR and clean up any remaining code for the v1beta1 version. Is it a lot of work to handle this for v1beta1 too?

Reply:
Not much work. Let me add the code for handling v1beta1 version as well.

Reply:
Added this.